dtfs-central-api/api-tests/helpers/with-validate-audit-details.api-tests.ts

```diff
@@ -0,0 +1,74 @@
+import { AuditDetails } from '@ukef/dtfs2-common/src/types/audit-details';
+import {
+  generatePortalAuditDetails,
+  generateSystemAuditDetails,
+  generateTfmAuditDetails,
+} from '@ukef/dtfs2-common/src/helpers/change-stream/generate-audit-details';
+import { MOCK_TFM_USER } from '../mocks/test-users/mock-tfm-user';
+import { MOCK_PORTAL_USER } from '../mocks/test-users/mock-portal-user';
+
+type MakeRequest = (auditDetails?: AuditDetails) => Promise<{ status: number; body: object }>;
+
+type Params = {
+  makeRequest: MakeRequest;
+  validUserTypes: ('tfm' | 'portal' | 'system')[];
+};
+
+export const withValidateAuditDetailsTests = ({ makeRequest, validUserTypes }: Params) => {
+  describe('when validating audit details', () => {
+    it('should return 400 if no auditDetails provided', async () => {
+      const { status, body } = await makeRequest();
+
+      expect(body).toEqual({ status: 400, message: 'Invalid auditDetails, Missing property `userType`' });
+      expect(status).toBe(400);
+    });
+
+    const { validAuditDetails, invalidAuditDetails } = getValidAndInvalidAuditDetails(validUserTypes);
+
+    withValidAuditDetailsTests(validAuditDetails, makeRequest)
+    if (invalidAuditDetails.length) {
+      withInvalidAuditDetailsTests(invalidAuditDetails, makeRequest);
+    }
+  });
+};
```

Comment:
I usually like to see the withs at the top -- like this :) just makes it easier to read in my eyes, lmk what you think

```suggestion
export const withValidateAuditDetailsTests = ({ makeRequest, validUserTypes }: Params) => {
  describe('when validating audit details', () => {
    const { validAuditDetails, invalidAuditDetails } = getValidAndInvalidAuditDetails(validUserTypes);

    withValidAuditDetailsTests(validAuditDetails, makeRequest)
    
    if (invalidAuditDetails.length) {
      withInvalidAuditDetailsTests(invalidAuditDetails, makeRequest);
    }
    
    it('should return 400 if no auditDetails provided', async () => {
      const { status, body } = await makeRequest();

      expect(body).toEqual({ status: 400, message: 'Invalid auditDetails, Missing property `userType`' });
      expect(status).toBe(400);
    });
  });
};
```